src/pybamm/util.py

```diff
@@ -101,13 +103,40 @@ def _find_matches(self, search_key: str, known_keys: list[str]):
             The term to search for in the keys.
         known_keys : list of str
             The list of known dictionary keys to search within.
-
+        min_similarity : float, optional
+            The minimum similarity threshold for a match.
+            Default is 0.4
         """
-        exact = [key for key in known_keys if search_key in key.lower()]
-        partial = difflib.get_close_matches(search_key, known_keys, n=5, cutoff=0.5)
-        return exact, partial
+        search_key = search_key.lower()
+        exact_matches = []
+        partial_matches = []
+
+        for key in known_keys:
+            key_lower = key.lower()
+            if search_key in key_lower:
+                key_words = key_lower.split()
+                significant_match = False
+
+                for word in key_words:
+                    similarity = difflib.SequenceMatcher(None, search_key, word).ratio()
 
-    def search(self, keys: str | list[str], print_values: bool = False):
+                    if similarity >= min_similarity:
+                        significant_match = True
+
+                if significant_match:
+                    exact_matches.append(key)
```

Comment:
Could we collapse this flow, as we mark `significant_match` as `True` if `if similarity >= min_similarity` is `True` in lines 123–124 and then we are checking for it again in lines 126–127?

```suggestion
                    if similarity >= min_similarity:
                        significant_match = True
                        exact_matches.append(key)
```